tests/phpunit/integration/Core/User_Surveys/REST_User_Surveys_ControllerTest.php

```diff
@@ -103,6 +117,145 @@ public function test_register_routes() {
 		$this->assertEquals( \WP_REST_Server::READABLE, $args[0]['methods'] );
 		$this->assertTrue( is_callable( $args[0]['callback'] ) );
 		$this->assertTrue( is_callable( $args[0]['permission_callback'] ) );
+
+		$args = $routes['survey']->get_args();
+		$this->assertEquals( 'core/user/data/survey', $routes['survey']->get_uri() );
+		$this->assertEquals( \WP_REST_Server::READABLE, $args[0]['methods'] );
+		$this->assertTrue( is_callable( $args[0]['callback'] ) );
+		$this->assertTrue( is_callable( $args[0]['permission_callback'] ) );
+	}
+
+	protected function send_request( $endpoint, $data, $response ) {
```

Comment:
The `$response` here is a bit confusing because it's used for internal (secondary) HTTP requests rather than the response from the controller's handler (which we of course wouldn't want to override here). Let's extract this to a separate method to make it more explicit that we're stubbing the response from a survey trigger on the service.